packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -628,16 +645,7 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
     if (chunkEvents != null) {
       chunkEvents.listen(
         (ImageChunkEvent event) {
-          if (hasListeners) {
-            // Make a copy to allow for concurrent modification.
-            final List<ImageChunkListener> localListeners = _listeners
-                .map<ImageChunkListener>((ImageStreamListener listener) => listener.onChunk)
-                .where((ImageChunkListener chunkListener) => chunkListener != null)
-                .toList();
-            for (final ImageChunkListener listener in localListeners) {
-              listener(event);
-            }
-          }
+          reportImageChunkEvent(event);
```

Comment:
You should be able to pass `reportImageChunkEvent` directly to `listen` without the need for an extra closure.